### PR TITLE
Bug 1418034 - Fix orpahaned configuration when update config parameters.

### PR DESCRIPTION
**Pull request by rubenvp8510**
@josejulio Could you please review,

This only mitigates the problem, there is still some orphaned configurations that appear from time to time, I'm not sure where that comes from.

### CODE REVIEW

File: modules/enterprise/server/jar/src/main/java/org/rhq/enterprise/server/configuration/ConfigurationManagerBean.java [hunk 1]
```diff
@@ -237,6 +238,7 @@ public void completePluginConfigurationUpdate(PluginConfigurationUpdate update)
             // Flush before merging to ensure the update has been persisted and avoid StaleStateExceptions.
             entityManager.flush();
             entityManager.merge(update);
+            entityManager.remove(oldPluginConfig);
```
Comment (loleary) [inline modules/enterprise/server/jar/src/main/java/org/rhq/enterprise/server/configuration/ConfigurationManagerBean.java]: Wouldn't we want the old plug-in configuration to remain in the plug-in config history page of the resource? It seems that a configuration update/merge would not result in an orphaned configuration object. Granted, if the configuration object has no associated configuration parameters, I would consider the update to be strange/broken.

File: modules/enterprise/server/jar/src/main/java/org/rhq/enterprise/server/configuration/ConfigurationManagerBean.java [hunk 2]
```diff
@@ -1568,7 +1570,9 @@ public void completeResourceConfigurationUpdate(ConfigurationUpdateResponse resp
         } else if (response.getStatus() == ConfigurationUpdateStatus.SUCCESS) {
             // link to the newer, persisted configuration object
             Resource resource = update.getResource();
+            Configuration oldResourceConfig = resource.getResourceConfiguration();
             resource.setResourceConfiguration(update.getConfiguration().deepCopyWithoutProxies());
+            entityManager.remove(oldResourceConfig);
```
Comment (loleary) [inline modules/enterprise/server/jar/src/main/java/org/rhq/enterprise/server/configuration/ConfigurationManagerBean.java]: Same question/concern as before. Wouldn't updated resource configuration simply result in the previous configuration history to appear on the resource's configuration history page?

**Comment by jshaughn** [comment]
LGTM

**Comment by josejulio** [comment]
I'm sure that this isn't something we want to do. This would result in deleting the previous configuration (which is not orpahaned AFAIK). 
Old configurations are displayed on the history, but with this change we always delete it.

screenshot from 2018-01-10 12-23-44

Maybe i'm missing something?

**Comment by rubenvp8510** [comment]
mm I'll review this in more detail, it seems like just in some situations the orphaned happens. but it happens for sure because I tested it.

**Comment by jshaughn** [comment]
For what it's worth, although there may be some simple explanation for some leaked config, it's more likely something very subtle in the entity mgmt.  More in line with the fixes done for BZ 1323332.  Possibly something to do with cascade delete.

**Comment by jshaughn** [comment]
@rubenvp8510 OK, these changes look good to me.  Adding the copy for the pluginConfig saved the history, right?   I think we just need to add the one other case we talked about.   For reference, other reviewers may want to see the comments in https://github.com/rubenvp8510/rhq/pull/1.

**Pull request merged**